test/test_all.py

```diff
@@ -0,0 +1,227 @@
+    timestamp_regex = re.compile(pattern=r"^[0-9]{4}/[0-9]{2}/[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}", flags=re.MULTILINE)
+    logs = re.sub(pattern=timestamp_regex, repl=timestamp_placeholder, string=logs)
+    golden_logs = re.sub(pattern=timestamp_regex, repl=timestamp_placeholder, string=golden_logs)
+
+    assert logs == golden_logs
+
+
+def check_index(configuration):
+    """Run tests to determine whether the generated library index is as expected.
+
+    Keyword arguments:
+    configuration -- dictionary defining the libraries-repository-engine configuration
+    """
+    # Load generated index
+    with pathlib.Path(configuration["LibrariesIndex"]).open(mode="r", encoding="utf-8") as library_index_file:
+        library_index = json.load(fp=library_index_file)
+    for release in library_index["libraries"]:
+        # The checksum values in the index will be different on every run, so it's necessary to remove them before
+        # comparing to the golden index
+        del release["checksum"]
+
+    # Load golden index
+    golden_library_index_template = test_data_path.joinpath("golden", "library_index.json").read_text(encoding="utf-8")
+    # Fill in mutable content
+    golden_library_index_string = string.Template(template=golden_library_index_template).substitute(
+        base_download_url=configuration["BaseDownloadUrl"]
+    )
+    golden_library_index = json.loads(golden_library_index_string)
+
+    # Order of releases in the index is arbitrary so a simply equality assertion is not possible
+    assert len(library_index["libraries"]) == len(golden_library_index["libraries"])
+    for release in library_index["libraries"]:
+        assert release in golden_library_index["libraries"]
+
+
+@pytest.fixture(scope="function")
+def run_command(pytestconfig, working_dir) -> typing.Callable[..., invoke.runners.Result]:
+    """Provide a wrapper around invoke's `run` API so that every test will work in the same temporary folder.
+
+    Useful reference:
+        http://docs.pyinvoke.org/en/1.4/api/runners.html#invoke.runners.Result
+    """
+
+    executable_path = pathlib.Path(pytestconfig.rootdir).parent / "libraries-repository-engine"
+
+    def _run(
+        cmd: list,
+        custom_working_dir: typing.Optional[str] = None,
+        custom_env: typing.Optional[dict] = None,
+    ) -> invoke.runners.Result:
+        if cmd is None:
+            cmd = []
+        if not custom_working_dir:
+            custom_working_dir = working_dir
+        quoted_cmd = []
+        for token in cmd:
+            quoted_cmd.append(f'"{token}"')
+        cli_full_line = '"{}" {}'.format(executable_path, " ".join(quoted_cmd))
+        run_context = invoke.context.Context()
+        # It might happen that we need to change directories between drives on Windows,
+        # in that case the "/d" flag must be used otherwise directory wouldn't change
+        cd_command = "cd"
+        if platform.system() == "Windows":
+            cd_command += " /d"
```

Comment:
Same consideration as above, with a container that replicates the production environment we could get rid of this "OS dependant" code